cfme/tests/intelligence/reports/test_reports_schedules.py

```diff
@@ -18,14 +44,19 @@ def schedule_files():
     return result
 
 
-@pytest.fixture(params=schedule_files(),
-                ids=[schedule.split(".")[0] for schedule in schedule_files()])
-def schedule_data(request):
+@pytest.fixture(
+    params=schedule_files(),
+    ids=[schedule.split(".")[0] for schedule in schedule_files()],
+)
+def schedule_data(request, interval=None):
```

Comment:
Given that this is a fixture, I don't think this is the correct pattern for interval.

I believe you need to define an `interval` fixture, with indirect parametrization on `test_schedule_queue`. That fixture needs to return `None` if request.param is not parametrized.

Reply:
Did a bit more testing, and confirmed my suspicions. 

This is actually not parametrizing interval at all. This can easily be observed by using a debugger here, or adding a pytest.skip() or fail() call and include the schedule data.

All of the parametrized test functions have the same timer - 2 hour, hourly.

Below is such output, with a `pytest.fail()` call in the schedule fixture between L61/62

```
___________________________________ ERROR at setup of test_schedule_queue[basic01-monthly] ___________________________________

schedule_data = {'active': True, 'description': 'description of testing schedule 1', 'email_options': {'send_csv': True, 'send_if_empty': True, 'send_pdf': True, 'send_txt': True}, 'emails': ['test1@example.test', 'test2@example.test'], ...}
appliance = IPAppliance.from_json('{"container": null, "ui_port": 443, "ssh_port": 22, "op..._dev": false, "version": "5.10.2.2", "db_host": null, "browser_steal": false}')

    @pytest.fixture(scope="function")
    def schedule(schedule_data, appliance):
>       pytest.fail('schedule_data: {}'.format(schedule_data))
E       Failed: schedule_data: {'name': 'testing schedule 1', 'timer': {'hours': '2 Hours', 'run': 'Hourly'}, 'from_email': 'sender@example.test', 'filter': ['Events', 'Policy', 'Policy Events for Last Week'], 'active': True, 'email_options': {'send_pdf': True, 'send_txt': True, 'send_if_empty': True, 'send_csv': True}, 'emails': ['test1@example.test', 'test2@example.test'], 'description': 'description of testing schedule 1'}

cfme/tests/intelligence/reports/test_reports_schedules.py:61: Failed
___________________________________ ERROR at setup of test_schedule_queue[basic01-hourly] ____________________________________

schedule_data = {'active': True, 'description': 'description of testing schedule 1', 'email_options': {'send_csv': True, 'send_if_empty': True, 'send_pdf': True, 'send_txt': True}, 'emails': ['test1@example.test', 'test2@example.test'], ...}
appliance = IPAppliance.from_json('{"container": null, "ui_port": 443, "ssh_port": 22, "op..._dev": false, "version": "5.10.2.2", "db_host": null, "browser_steal": false}')

    @pytest.fixture(scope="function")
    def schedule(schedule_data, appliance):
>       pytest.fail('schedule_data: {}'.format(schedule_data))
E       Failed: schedule_data: {'name': 'testing schedule 1', 'timer': {'hours': '2 Hours', 'run': 'Hourly'}, 'from_email': 'sender@example.test', 'filter': ['Events', 'Policy', 'Policy Events for Last Week'], 'active': True, 'email_options': {'send_pdf': True, 'send_txt': True, 'send_if_empty': True, 'send_csv': True}, 'emails': ['test1@example.test', 'test2@example.test'], 'description': 'description of testing schedule 1'}

cfme/tests/intelligence/reports/test_reports_schedules.py:61: Failed
____________________________________ ERROR at setup of test_schedule_queue[basic01-once] _____________________________________

schedule_data = {'active': True, 'description': 'description of testing schedule 1', 'email_options': {'send_csv': True, 'send_if_empty': True, 'send_pdf': True, 'send_txt': True}, 'emails': ['test1@example.test', 'test2@example.test'], ...}
appliance = IPAppliance.from_json('{"container": null, "ui_port": 443, "ssh_port": 22, "op..._dev": false, "version": "5.10.2.2", "db_host": null, "browser_steal": false}')

    @pytest.fixture(scope="function")
    def schedule(schedule_data, appliance):
>       pytest.fail('schedule_data: {}'.format(schedule_data))
E       Failed: schedule_data: {'name': 'testing schedule 1', 'timer': {'hours': '2 Hours', 'run': 'Hourly'}, 'from_email': 'sender@example.test', 'filter': ['Events', 'Policy', 'Policy Events for Last Week'], 'active': True, 'email_options': {'send_pdf': True, 'send_txt': True, 'send_if_empty': True, 'send_csv': True}, 'emails': ['test1@example.test', 'test2@example.test'], 'description': 'description of testing schedule 1'}

cfme/tests/intelligence/reports/test_reports_schedules.py:61: Failed
____________________________________ ERROR at setup of test_schedule_queue[basic01-daily] ____________________________________

schedule_data = {'active': True, 'description': 'description of testing schedule 1', 'email_options': {'send_csv': True, 'send_if_empty': True, 'send_pdf': True, 'send_txt': True}, 'emails': ['test1@example.test', 'test2@example.test'], ...}
appliance = IPAppliance.from_json('{"container": null, "ui_port": 443, "ssh_port": 22, "op..._dev": false, "version": "5.10.2.2", "db_host": null, "browser_steal": false}')

    @pytest.fixture(scope="function")
    def schedule(schedule_data, appliance):
>       pytest.fail('schedule_data: {}'.format(schedule_data))
E       Failed: schedule_data: {'name': 'testing schedule 1', 'timer': {'hours': '2 Hours', 'run': 'Hourly'}, 'from_email': 'sender@example.test', 'filter': ['Events', 'Policy', 'Policy Events for Last Week'], 'active': True, 'email_options': {'send_pdf': True, 'send_txt': True, 'send_if_empty': True, 'send_csv': True}, 'emails': ['test1@example.test', 'test2@example.test'], 'description': 'description of testing schedule 1'}

cfme/tests/intelligence/reports/test_reports_schedules.py:61: Failed
___________________________________ ERROR at setup of test_schedule_queue[basic01-weekly] ____________________________________

schedule_data = {'active': True, 'description': 'description of testing schedule 1', 'email_options': {'send_csv': True, 'send_if_empty': True, 'send_pdf': True, 'send_txt': True}, 'emails': ['test1@example.test', 'test2@example.test'], ...}
appliance = IPAppliance.from_json('{"container": null, "ui_port": 443, "ssh_port": 22, "op..._dev": false, "version": "5.10.2.2", "db_host": null, "browser_steal": false}')

    @pytest.fixture(scope="function")
    def schedule(schedule_data, appliance):
>       pytest.fail('schedule_data: {}'.format(schedule_data))
E       Failed: schedule_data: {'name': 'testing schedule 1', 'timer': {'hours': '2 Hours', 'run': 'Hourly'}, 'from_email': 'sender@example.test', 'filter': ['Events', 'Policy', 'Policy Events for Last Week'], 'active': True, 'email_options': {'send_pdf': True, 'send_txt': True, 'send_if_empty': True, 'send_csv': True}, 'emails': ['test1@example.test', 'test2@example.test'], 'description': 'description of testing schedule 1'}

cfme/tests/intelligence/reports/test_reports_schedules.py:61: Failed
====================================================== warnings summary ======================================================
None
  Module already imported so cannot be rewritten: cfme

-- Docs: http://doc.pytest.org/en/latest/warnings.html
===================================================== 0 tests deselected =====================================================
============================================ 1 warnings, 5 error in 70.31 seconds ============================================
```

Reply:
This is horrible :roll_eyes: 
_/me :hides_face:_
I need to start paying more attention when my tests run :woman_facepalming: 

I'll get started on fixing this.